frontend/public/components/instantiate-template.tsx

```diff
@@ -129,55 +129,39 @@ const stateToProps = (state: RootState) => ({
   models: state.k8s.getIn(['RESOURCES', 'models']),
 });
 
-class TemplateForm_ extends React.Component<
-  TemplateFormProps & WithTranslation,
-  TemplateFormState
-> {
-  constructor(props: TemplateFormProps & WithTranslation) {
-    super(props);
+const TemplateForm_: React.FC<TemplateFormProps> = (props) => {
+  const { preselectedNamespace: ns = '', activePerspective, perspectiveExtensions, obj } = props;
 
-    const { preselectedNamespace: namespace = '' } = this.props;
-    const parameters = this.getParameterValues(props);
-    this.state = {
-      namespace,
-      parameters,
-      inProgress: false,
-      error: '',
-    };
-  }
-
-  componentDidUpdate(prevProps: TemplateFormProps & WithTranslation) {
-    if (this.props.obj.data?.parameters !== prevProps.obj.data?.parameters) {
-      const parameters = this.getParameterValues(this.props);
-      this.setState({ parameters });
-    }
-  }
-
-  getParameterValues = (props: TemplateFormProps & WithTranslation) => {
-    const templateParameters: TemplateParameter[] = props.obj.data.parameters || [];
+  const getParameterValues = (input: TemplateFormProps) => {
```

Comment:
FYI, this function already has access to `props` and its destructured elements via [closure](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures) so you could omit the `input` parameter here.

```ts
const templateParameters: TemplateParameter[] = obj.data.parameters || [];
```